corehq/apps/domain/decorators.py

```diff
@@ -178,6 +179,16 @@ def login_or_basic_ex(allow_cc_users=False):
 login_or_basic = login_or_basic_ex()
 
 
+def login_or_digest_or_basic(fn):
+    @wraps(fn)
+    def _inner(request, *args, **kwargs):
+        return {
+            'basic': login_or_basic_ex(allow_cc_users=True),
+            'digest': login_or_digest_ex(allow_cc_users=True),
+        }[determine_authtype(request)](fn)(request, *args, **kwargs)
```

Comment:
This is a bit of a nitpick, but if `determine_authtype` and `guess_phone_type_from_user_agent` are going to be used more globally like this, I'd prefer not having them stay in `corehq.apps.receiverwrapper.util`.

Additionally, `determine_authtype` pulls from the `authtype` GET param, so `?authtype=digest` will override the guessing, and `?authtype=yourmom` will make the URL 500 with a `KeyError`. I'm not sure either of those is specifically desired, so it seems wrong to have those behaviors get roped in as well.